pkg/server/server.go

```diff
@@ -89,7 +91,13 @@ type ExtraConfig struct {
 	// resource identities into the rest.Config used by the client. Only after it succeeds,
 	// the clients can wildcard-list/watch most kcp resources.
 	resolveIdentities func(ctx context.Context) error
-	identityConfig    *rest.Config
+
+	// ResolveRootKcpShardIdentities is to be called on server start until it succeeds. It injects the kcp
+	// resource identities for the root shard into the rest.Config used by the client.
+	// Only after it succeeds, the clients can wildcard-list/watch most kcp resources.
+	ResolveRootKcpShardIdentities func(ctx context.Context) error
+
+	identityConfig *rest.Config
```

Comment:
am still confused about the new fields. The existing should do exactly that.

Reply:
ah, I think I know what you mean :)

Reply:
I think this is what we need https://github.com/kcp-dev/kcp/pull/1620

Reply:
can you wire it here?

Reply:
done